lib/matplotlib/axes/_axes.py

```diff
@@ -5695,8 +5695,7 @@ def _pcolorargs(self, funcname, *args, shading='auto', **kwargs):
                     Y = Y.data
             nrows, ncols = C.shape
         else:
-            raise TypeError(f'{funcname}() takes 1 or 3 positional arguments '
-                            f'but {len(args)} were given')
+            raise _api.nargs_error(funcname, "1 or 3", len(args))
```

Comment:
I find it often hard to read/understand such message-generating functions where you have to pass in many parts to plug into a message.

We *could* taylor the parameter names to make the call more like the message to be generated, i.e.

```suggestion
            raise _api.nargs_error(
                funcname, takes="1 or 3", but_given=len(args))
```

Not sure if that's too cute.

Reply:
sure, done